src/memefactory/server/syncer.cljs

```diff
@@ -109,178 +111,180 @@
-               #(db/upsert-user! {:user/address voter}))))
+  (safe-go
+   (let [{:keys [:registry-entry :timestamp :version :voter :option]} args
+         vote (db/get-vote {:reg-entry/address registry-entry :vote/voter voter} [:vote/voter :vote/amount :reg-entry/address])
+         re (db/get-registry-entry {:reg-entry/address registry-entry} [:challenge/votes-against :challenge/votes-for :challenge/votes-total])
+         {:keys [:challenge/votes-total :challenge/votes-against :challenge/votes-for]} re]
+     (let [option (bn/number option)
+           {:keys [:vote/voter :vote/amount :vote/option] :as vote} (merge vote
+                                                                           {:vote/option option
+                                                                            :vote/revealed-on timestamp})
+           errors (vote-sanity-check vote)]
+       (if-not (empty? (:errors errors))
+         errors
+
+         (do (db/update-registry-entry! (cond-> {:reg-entry/address registry-entry
+                                                 :reg-entry/version version}
+
+                                          true
+                                          (assoc :challenge/votes-total (+ votes-total amount))
+
+                                          (= (vote-options option) :vote.option/vote-against)
+                                          (assoc :challenge/votes-against (+ votes-against amount))
+
+                                          (= (vote-options option) :vote.option/vote-for)
+                                          (assoc :challenge/votes-for (+ votes-for amount))))
+             (db/update-vote! vote)))
+       (db/upsert-user! {:user/address voter})))))
 
 (defn vote-reward-claimed-event [_ {:keys [:args]}]
-  (let [{:keys [:registry-entry :timestamp :version :voter :amount]} args]
-    (promise-> (js/Promise.resolve (db/update-vote! {:reg-entry/address registry-entry
-                                                     :vote/voter voter
-                                                     :vote/claimed-reward-on timestamp}))
-               #(db/inc-user-field! voter :user/voter-total-earned (bn/number amount)))))
+  (safe-go
+   (let [{:keys [:registry-entry :timestamp :version :voter :amount]} args]
+     (db/update-vote! {:reg-entry/address registry-entry
+                       :vote/voter voter
+                       :vote/claimed-reward-on timestamp})
+     (db/inc-user-field! voter :user/voter-total-earned (bn/number amount)))))
 
 (defn vote-amount-reclaimed-event [_ {:keys [:args]} ]
   (let [{:keys [:registry-entry :timestamp :version :voter :amount]} args]
-    (js/Promise.resolve (db/update-vote! {:reg-entry/address registry-entry
-                                          :vote/voter voter
-                                          :vote/reclaimed-amount-on timestamp}))))
+    (db/update-vote! {:reg-entry/address registry-entry
```

Comment:
not wrapped in macro?

Reply:
Same as the previous, it doesn't need it since it doesn't contain any async call in its body